### ISSUE TITLE
Investigate auto generating docs by adding more information to the yaml

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#277]
## Specification

See if it makes sense to add additional fields to the yaml config. We could potentially add a docs_description field that included markdown describing how to use the cop. Ideally we end up with content that is at a minimum as good as what we had on foodcritic.io.

The current data available for each rule is similar to this:

```yml
Chef/LocaleDeprecatedLcAllProperty:
  Description: The local resource's lc_all property has been deprecated and will be
    removed in Chef Infra Client 16
  Enabled: true
  VersionAdded: 5.5.0
  Exclude:
  - "/Users/tsmith/dev/work/cookstyle/**/metadata.rb"
```

We could certainly generate docs with that information, and it would be better than nothing, but I think we can do a lot better.

## Downstream Impact

The docs team will most likely consume this information to populate the docs site. Obviously this is going to require a lot of coordination so we can figure out how to best meet both of our needs.

**Comment by kagarmoe** [comment]
I'm always up for automating the docs! I have a few quick notes:
- Do we want a `short_description` field for the cli output and a `long_description` field for the docs?
- Do we want a field describing the impact for the user, perhaps an InSpec-like scale from the lines of "poorly expressed, but works" to "will break your world"?
- Do we want a field for explaining how to change the problem?

**Comment by tas50** [comment]
The basic part of this is shipped in #278

**Comment by tas50** [comment]
We get the description field, but we also get the examples section pumped into the doc. I'm going to work a bit on the format, but it's a good step in the right direction. It may or may not be something we can use on the hugo docs site in the future. We'll have to see where it goes.
